### PR TITLE
Fix competency extraction feature

### PR DESCRIPTION
**Pull request by MichaelOwenDyer**
## Summary by CodeRabbit

- **New Features**
	- Introduced a competency extraction pipeline to enhance the processing of course descriptions.
	- Added new API endpoints for running competency extraction asynchronously.
	- Implemented a structured AI prompt for extracting competencies from course descriptions.
	- Introduced a new class for managing competency extraction status updates.
	- Enhanced data validation capabilities with new competency data models.
	- Simplified state management in the status update system for competency extraction.
  
- **Bug Fixes**
	- Streamlined data structures in various data transfer objects for improved clarity and performance.

- **Documentation**
	- Enhanced docstring readability in the course chat pipeline without altering functionality.

Some manual notes:
- I decided not to use LangChain to implement this feature, I just call the CapabilityRequestHandler directly
- The original implementation of this feature called the LLM in a loop for each additional competency, which was horrible for performance. Now, I just call the LLM once and parse many competencies out of the returned string. In my testing, the pipeline takes only a second or two to complete now.
- This PR contains only two changes to existing code: I de-duplicated the `settings` and `initialStages` attributes out of the subclasses of `PipelineExecutionDTO` into the superclass. This allows me to use `PipelineExecutionDTO` in my new `CompetencyExtractionPipelineExecutionDTO` (I opted for composition, not inheritance)

### CODE REVIEW

**Review by yassinsws — DISMISSED** [review]
Code looks Good

File: app/web/status/status_update.py [hunk 1]
```diff
@@ -5,6 +5,9 @@
 import requests
 from abc import ABC
 
+from domain.status.competency_extraction_status_update_dto import (
```
Comment (yassinsws) [inline app/web/status/status_update.py:8]: replace with 
from ...domain.status.competency_extraction_status_update_dto import (

Comment (MichaelOwenDyer) [reply app/web/status/status_update.py:8]: You smart man this was exactly the reason why generation was failing

**Review by bassner — APPROVED** [review]
Code LGTM

**Pull request merged**